### ISSUE TITLE
ALMA-Fix: `replace_all("hbzId","\\(DE-605\\)(.*)","$1")` breaks due to Expected String, got Array

### ISSUE THREAD

**Issue by TobiasNx** [hbz/lobid-resources#1377]
990128175850206441 breaks.

```
[WARNING] 
org.metafacture.metafix.FixExecutionException: Error while executing Fix expression (at file:/home/tobias/git/lobid-resources/src/main/resources/alma/fix/fields.fix, line 15): replace_all("hbzId","\\(DE-605\\)(.*)","$1")
...
Caused by: java.lang.IllegalStateException: Expected String, got Array
```

**Comment by TobiasNx** [comment]
```
2022-08-11 10:51:15 INFO [org.lobid.resources.ElasticsearchIndexer.onSetReceiver(ElasticsearchIndexer.java:185)] - Set index.refresh_interval to -1
 INFO [AlmaMarcXmlFix2lobidJsonEs] (ObjectThreader.java:73) - Adding thread 1
v ,. INFO [Thread-2] (Metafix.java:404) - Current record: {DEL-1={a=Y}, leader=00912djm#a2200301#cc4500, 005=20220810154214.0, 007=ss#|||||||||||, 008=030312|2002####ru#|||##############rus#c, 001=990003881000107476, 0167 ={a=1072986760, 2=OCoLC}, 020  ={a=3875482972, 9=3-87548-297-2, 0=(uri) http://www.isbnsearch.org/isbn/3875482972}, 035  =[{a=(DE-605)TT000421960}, {a=(OCoLC)1072986760, 0=(uri) http://www.worldcat.org/oclc/1072986760}, {a=(DE-599)HBZTT000421960}, {a=(DE-605)990128175850206441}], 040  ={a=Lan 1, b=ger, c=HBZ, d=[HBZ/ALMA-NZ, DE-605], e=rakwb}, 041  ={a=rus}, 044  ={c=XA-RU}, 090  ={b=b}, 24500={a=Sovremennaja russkaja idiomatika, n=Toncass, h=Tonträger, c=Elena Minakova}, 264 1={a=Hamburg, b=Buske, c=[2002]}, 300  ={a=Tonkassette (90 Min.)}, 77308={w=(DE-605)HT013163888, q=Toncass.}, 7001 ={a=Minakova, Elena E., 4=oth, 9=O:H}, 9640i={I=marc-FMT via MAB2MARC, F=FMT, A=MU}, 9640s=[{F=030, A=a|1uc||||||||}, {F=050, A=|||||ad|||||||}], 991  ={a=Gc, 9=local}, H528 ={b=RL002, c=AUSLD, h=siehe Überordnung, 8=22106034740007476}, HOL  ={d=2022-08-05 13:50:09, 8=22106034740007476, b=2022-07-12 11:49:51, M=49HBZ_RTU, g=false, a=import, c=System}, almaMmsId=990003881000107476, id=https://lobid.org/resources/990003881000107476#!, @context=https://lobid.org/resources/context.jsonld, @title=Sovremennaja russkaja idiomatika, @titleOfSubSeries_n=Toncass, @titleOfSubSeries=, Toncass, titleOfSubSeries=Toncass, title=Sovremennaja russkaja idiomatika, Toncass, alternativeTitle[]=[], otherTitleInformation[]=[], edition[]=[], responsibilityStatement[]=[Elena Minakova], contribution[]=[{agent={label=Minakova, Elena E., type[]=[Person], role={id=http://id.loc.gov/vocabulary/relators/oth}}, type[]=[Contribution]}], describedBy={id=http://lobid.org/resources/990003881000107476, label=Webseite der hbz-Ressource 990003881000107476, type[]=[BibliographicDescription], inDataset={id=http://lobid.org/resources/dataset#!, label=lobid-resources – Der hbz-Verbundkatalog als Linked Open Data}, resultOf={type[]=[CreateAction], endTime=0000-00-00T00:00:00, instrument={id=https://github.com/hbz/lobid-resources, type[]=[SoftwareApplication], label=Software lobid-resources}, object={id=https://lobid.org/hbz01/990003881000107476, type[]=[DataFeedItem], label=hbz-Ressource 990003881000107476 im Exportformat MARC21 XML, inDataset={id=https://datahub.io/dataset/hbz_unioncatalog, label=hbz_unioncatalog}}}, license[]=[{id=http://creativecommons.org/publicdomain/zero/1.0, label=Creative Commons-Lizenz CC0 1.0 Universal}], sourceOrganization={id=http://lobid.org/organisations/DE-Lan 1#!, label=lobid Organisation}, provider={id=http://lobid.org/organisations/DE-HBZ#!, label=lobid Organisation}, modifiedBy={id=http://lobid.org/organisations/DE-HBZ/ALMA-NZ#!, label=lobid Organisation}}, hbzId=[(DE-605)TT000421960, (DE-605)990128175850206441]}
 INFO [Thread-2] (Metafix.java:404) - Current record: {DEL-1={a=Y}, leader=00912djm#a2200301#cc4500, 005=20220810154214.0, 007=ss#|||||||||||, 008=030312|2002####ru#|||##############rus#c, 001=990003881000107476, 0167 ={a=1072986760, 2=OCoLC}, 020  ={a=3875482972, 9=3-87548-297-2, 0=(uri) http://www.isbnsearch.org/isbn/3875482972}, 035  =[{a=(DE-605)TT000421960}, {a=(OCoLC)1072986760, 0=(uri) http://www.worldcat.org/oclc/1072986760}, {a=(DE-599)HBZTT000421960}, {a=(DE-605)990128175850206441}], 040  ={a=Lan 1, b=ger, c=HBZ, d=[HBZ/ALMA-NZ, DE-605], e=rakwb}, 041  ={a=rus}, 044  ={c=XA-RU}, 090  ={b=b}, 24500={a=Sovremennaja russkaja idiomatika, n=Toncass, h=Tonträger, c=Elena Minakova}, 264 1={a=Hamburg, b=Buske, c=[2002]}, 300  ={a=Tonkassette (90 Min.)}, 77308={w=(DE-605)HT013163888, q=Toncass.}, 7001 ={a=Minakova, Elena E., 4=oth, 9=O:H}, 9640i={I=marc-FMT via MAB2MARC, F=FMT, A=MU}, 9640s=[{F=030, A=a|1uc||||||||}, {F=050, A=|||||ad|||||||}], 991  ={a=Gc, 9=local}, H528 ={b=RL002, c=AUSLD, h=siehe Überordnung, 8=22106034740007476}, HOL  ={d=2022-08-05 13:50:09, 8=22106034740007476, b=2022-07-12 11:49:51, M=49HBZ_RTU, g=false, a=import, c=System}, almaMmsId=990003881000107476, id=https://lobid.org/resources/990003881000107476#!, @context=https://lobid.org/resources/context.jsonld, @title=Sovremennaja russkaja idiomatika, @titleOfSubSeries_n=Toncass, @titleOfSubSeries=, Toncass, titleOfSubSeries=Toncass, title=Sovremennaja russkaja idiomatika, Toncass, alternativeTitle[]=[], otherTitleInformation[]=[], edition[]=[], responsibilityStatement[]=[Elena Minakova], contribution[]=[{agent={label=Minakova, Elena E., type[]=[Person], role={id=http://id.loc.gov/vocabulary/relators/oth}}, type[]=[Contribution]}], describedBy={id=http://lobid.org/resources/990003881000107476, label=Webseite der hbz-Ressource 990003881000107476, type[]=[BibliographicDescription], inDataset={id=http://lobid.org/resources/dataset#!, label=lobid-resources – Der hbz-Verbundkatalog als Linked Open Data}, resultOf={type[]=[CreateAction], endTime=0000-00-00T00:00:00, instrument={id=https://github.com/hbz/lobid-resources, type[]=[SoftwareApplication], label=Software lobid-resources}, object={id=https://lobid.org/hbz01/990003881000107476, type[]=[DataFeedItem], label=hbz-Ressource 990003881000107476 im Exportformat MARC21 XML, inDataset={id=https://datahub.io/dataset/hbz_unioncatalog, label=hbz_unioncatalog}}}, license[]=[{id=http://creativecommons.org/publicdomain/zero/1.0, label=Creative Commons-Lizenz CC0 1.0 Universal}], sourceOrganization={id=http://lobid.org/organisations/DE-Lan 1#!, label=lobid Organisation}, provider={id=http://lobid.org/organisations/DE-HBZ#!, label=lobid Organisation}, modifiedBy={id=http://lobid.org/organisations/DE-HBZ/ALMA-NZ#!, label=lobid Organisation}}, hbzId=[(DE-605)TT000421960, (DE-605)990128175850206441]}
[WARNING] 
org.metafacture.metafix.FixExecutionException: Error while executing Fix expression (at file:/home/tobias/git/lobid-resources/src/main/resources/alma/fix/fields.fix, line 15): replace_all("hbzId","\\(DE-605\\)(.*)","$1")
    at org.metafacture.metafix.RecordTransformer.tryRun (RecordTransformer.java:244)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$2 (RecordTransformer.java:123)
    at java.lang.Iterable.forEach (Iterable.java:75)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:122)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:114)
    at org.metafacture.metafix.FixMethod$1.apply (FixMethod.java:60)
    at org.metafacture.metafix.RecordTransformer.lambda$processFunction$20 (RecordTransformer.java:204)
    at org.metafacture.metafix.RecordTransformer.lambda$processFix$22 (RecordTransformer.java:224)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$1 (RecordTransformer.java:123)
    at org.metafacture.metafix.RecordTransformer.tryRun (RecordTransformer.java:235)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$2 (RecordTransformer.java:123)
    at java.lang.Iterable.forEach (Iterable.java:75)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:122)
    at org.metafacture.metafix.Metafix.endRecord (Metafix.java:200)
    at org.metafacture.biblio.marc21.MarcXmlHandler.endElement (MarcXmlHandler.java:132)
    at org.apache.xerces.parsers.AbstractSAXParser.endElement (Unknown Source)
    at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanEndElement (Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch (Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument (Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse (Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse (Unknown Source)
    at org.apache.xerces.parsers.XMLParser.parse (Unknown Source)
    at org.apache.xerces.parsers.AbstractSAXParser.parse (Unknown Source)
    at org.metafacture.xml.XmlDecoder.process (XmlDecoder.java:69)
    at org.metafacture.xml.XmlDecoder.process (XmlDecoder.java:43)
    at org.metafacture.strings.StringReader.process (StringReader.java:48)
    at org.metafacture.strings.StringReader.process (StringReader.java:34)
    at org.metafacture.flowcontrol.ObjectPipeDecoupler$Feeder.run (ObjectPipeDecoupler.java:175)
    at java.lang.Thread.run (Thread.java:829)
Caused by: java.lang.IllegalStateException: Expected String, got Array
    at org.metafacture.metafix.Value$TypeMatcher.lambda$orElseThrow$0 (Value.java:342)
    at org.metafacture.metafix.Value$TypeMatcher.orElse (Value.java:335)
    at org.metafacture.metafix.Value$TypeMatcher.orElseThrow (Value.java:340)
    at org.metafacture.metafix.Value.lambda$asString$2 (Value.java:180)
    at org.metafacture.metafix.Value.extractType (Value.java:212)
    at org.metafacture.metafix.Value.asString (Value.java:180)
    at org.metafacture.metafix.Value$Hash.getField (Value.java:627)
    at org.metafacture.metafix.Value$Hash.get (Value.java:612)
    at org.metafacture.metafix.FixPath.findIn (FixPath.java:58)
    at org.metafacture.metafix.Record.transform (Record.java:201)
    at org.metafacture.metafix.FixMethod$39.apply (FixMethod.java:514)
    at org.metafacture.metafix.RecordTransformer.lambda$processFunction$20 (RecordTransformer.java:204)
    at org.metafacture.metafix.RecordTransformer.lambda$processFix$22 (RecordTransformer.java:224)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$1 (RecordTransformer.java:123)
    at org.metafacture.metafix.RecordTransformer.tryRun (RecordTransformer.java:235)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$2 (RecordTransformer.java:123)
    at java.lang.Iterable.forEach (Iterable.java:75)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:122)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:114)
    at org.metafacture.metafix.FixMethod$1.apply (FixMethod.java:60)
    at org.metafacture.metafix.RecordTransformer.lambda$processFunction$20 (RecordTransformer.java:204)
    at org.metafacture.metafix.RecordTransformer.lambda$processFix$22 (RecordTransformer.java:224)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$1 (RecordTransformer.java:123)
    at org.metafacture.metafix.RecordTransformer.tryRun (RecordTransformer.java:235)
    at org.metafacture.metafix.RecordTransformer.lambda$transform$2 (RecordTransformer.java:123)
    at java.lang.Iterable.forEach (Iterable.java:75)
    at org.metafacture.metafix.RecordTransformer.transform (RecordTransformer.java:122)
    at org.metafacture.metafix.Metafix.endRecord (Metafix.java:200)
    at org.metafacture.biblio.marc21.MarcXmlHandler.endElement (MarcXmlHandler.java:132)
    at org.apache.xerces.parsers.AbstractSAXParser.endElement (Unknown Source)
    at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanEndElement (Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScan  annerImpl$FragmentContentDispatcher.dispatch (Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument (Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse (Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse (Unknown Source)
    at org.apache.xerces.parsers.XMLParser.parse (Unknown Source)
    at org.apache.xerces.parsers.AbstractSAXParser.parse (Unknown Source)
    at org.metafacture.xml.XmlDecoder.process (XmlDecoder.java:69)
    at org.metafacture.xml.XmlDecoder.process (XmlDecoder.java:43)
    at org.metafacture.strings.StringReader.process (StringReader.java:48)
    at org.metafacture.strings.StringReader.process (StringReader.java:34)
    at org.metafacture.flowcontrol.ObjectPipeDecoupler$Feeder.run (ObjectPipeDecoupler.java:175)
    at java.lang.Thread.run (Thread.java:829)
```

I tested it locally with a singular testfile and I cannot recreated the above error.

**Comment by TobiasNx** [comment]
Solved by #1378
